Restructure and rename admin controller

michaelKaefer opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Rename and restructure according to https://github.com/PrestaShop/prestafony-project/blob/master/docs/migration-guide.md under headline "Modern/Symfony Controllers"
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -
| How to test?  | -

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/9159)
<!-- Reviewable:end -->

sLorenzini committed:
Clean header backoffice design - more clear for merchant

sLorenzini committed:
Clean header backoffice design - more clear for merchant #2

sLorenzini committed:
Improve Header BO design

unlocomqx committed:
CO: Add customization ID to cart ajax response and updateCart event data

sLorenzini committed:
Update error material-icon on BO header

sLorenzini committed:
Improve material-icon on BO header

sLorenzini committed:
Improve class material-icons on BO header

sLorenzini committed:
Improve header links on default BO theme

sLorenzini committed:
Improve header links on new BO theme

michaelKaefer committed:
remove call to Tools::array_replace()

michaelKaefer committed:
Refactor Dispatcher::useDefaultController()

Cleaner and more testable code

michaelKaefer committed:
Refactor Dispatcher::useDefaultController()

Cleaner and more testable code

michaelKaefer committed:
Refactor Dispatcher::useDefaultController()

Cleaner and more testable code

Quetzacoalt91 committed:
Add dockerfile for single container runner

sarjon committed:
start working on localization migration

rdy4ever committed:
Fix phone filed type

In the customer address form, the phone filed should have the type set to 'tel'. This way, when completing the filed on a smartphone, a specific phone keyboard will be used, making the input of the phone easier.

rdy4ever committed:
use elseif instead of if

fatmaBouchekoua committed:
TE:Add export shopping carts scenario

fouratachour committed:
TE: change the order of the scenarios

andypieters committed:
Check if key exists before checking the value

Commit:
finish first part of localization configuration

Commit:
migrate local units configuration

Commit:
migrate advanced localization settings

Commit:
split localization template into separate blocks

Commit:
handle default currency & language data update when saving configuration

Commit:
wip import localization pack

PierreRambaud committed:
Migrate Orders > Delivery > Slip

First commit for the migration of AdminDeliverySlipController to a
Symfony page.

PierreRambaud committed:
Add translations

PierreRambaud committed:
Continue with pdf generator form

Commit:
Casting preservedGifts id_product_attribute to int
To prevent errors when id_product_attribute is null

sarjon committed:
implement localization pack loader

eternoendless committed:
Fix navbar height

eternoendless committed:
Fix header on default theme

+ Remove icons from quick links

sarjon committed:
add localization pack downloader

sarjon committed:
fix license header

PierreRambaud committed:
Fix datepicker

mickaelandrieu committed:
Reorganize routing of Back Office

mickaelandrieu committed:
Fixed url of Stocks action page (hardcoded url)

mickaelandrieu committed:
build assets

PierreRambaud committed:
Generate pdf + fix datepicker

PierreRambaud committed:
Add unit tests

PierreRambaud committed:
Fix wrong namespace

PierreRambaud committed:
Comments + remove useless changes

PierreRambaud committed:
Remove incomplete test

PierreRambaud committed:
wrong key for flashbag

sarjon committed:
create localization pack importer

Commit:
improve localization pack import

Commit:
improve localization pack choices loading

fatmaBouchekoua committed:
TE: Navigate between pages in catalog page

PierreRambaud committed:
Use adapter instead of legacy classes

PierreRambaud committed:
Sorted lines

Commit:
add check for country iso code when importing localization pack

Commit:
add demo mode & permissions check

Commit:
clear module list cache when default country has changed

Commit:
add warning when currency is changed

Commit:
remove legacy code & templates

Commit:
update route names & Link class

mickaelandrieu committed:
Refactored API part of routing

Commit:
update docs

PierreRambaud committed:
Merge branch 'develop' of github.com:PrestaShop/PrestaShop into improvement/delivery-slip

mickaelandrieu committed:
Refactored Advanced parameters section

PierreRambaud committed:
Fix wrong package-lock file

mickaelandrieu committed:
Refactored Shop Parameters section

mickaelandrieu committed:
Coding styles

PierreRambaud committed:
Regression with DatePicker and checkboxes on stock page

DatePicker field not working due to wrong css selector.
And because it can be on every page, run datepicker function
directly on theme.js. Fix problem where we can't check checkboxes
on the stock page.

PierreRambaud committed:
Commit assets

Commit:
start working on geolocation migration

Commit:
address @PierreRambaud and @mickaelandrieu review

Commit:
update language activator method naming

PierreRambaud committed:
Make  SlipConfiguration as final class

PierreRambaud committed:
Remove useless part of code

PierreRambaud committed:
address @mickaelandrieu review

PierreRambaud committed:
Merge branch 'develop' of github.com:PrestaShop/PrestaShop into improvement/delivery-slip

PierreRambaud committed:
Bad wording

PierreRambaud committed:
Use just one isset

PierreRambaud committed:
address @mickaelandrieu reviewb

PierreRambaud committed:
address @mickaelandrieu reviewb

Quetzacoalt91 committed:
Fix undefined method trans on AddonsController

Commit:
add data configurators, form data provider

michaelKaefer committed:
Migrate AdminThemesCatalogController to ThemeController

michaelKaefer committed:
Merge remote-tracking branch 'upstream/develop' into develop

michaelKaefer committed:
Changes after very helpful review

michaelKaefer committed:
Remove @author - even if it seems to be common at Symfony

eternoendless committed:
Fix bugs and responsiveness on the BO default theme

mickaelandrieu committed:
Merge pull request #9151 from Quetzacoalt91/addons-ctrl-trans

Fix undefined method trans() when loggin on the marketplace from the module page

PierreRambaud committed:
Missing orderBy and sortOrder in template

PierreRambaud committed:
Merge branch 'improvement/delivery-slip' of github.com:PierreRambaud/PrestaShop into improvement/delivery-slip

eternoendless committed:
Use ems instead of rems for responsive breakpoints

Quetzacoalt91 committed:
Merge pull request #9157 from PierreRambaud/fix/boom-5652

State filter in catalog page doesn't work

Commit:
finish configuration save, extract geolocation code

eternoendless committed:
Fix display of employee icon when mobile

eternoendless committed:
Update to UI kit 1.0.1

Used NPM 6, hence the big diff in package-lock

eternoendless committed:
Fixed header on the new theme

eternoendless committed:
Fixed selector in E2E tests

michaelKaefer committed:
Fix bugs of previous commit

michaelKaefer committed:
Remove mpty line

michaelKaefer committed:
Make version part of URL dynamic

eternoendless committed:
Fixed multishop menu style (a little)

michaelKaefer committed:
Rename to ThemeCatalogController

michaelKaefer committed:
Rename to ThemeCatalogController

michaelKaefer committed:
Rename file

michaelKaefer committed:
Merge remote-tracking branch 'upstream/develop' into develop

michaelKaefer committed:
Bug fix

michaelKaefer committed:
Restructure and rename admin controller aread for modules catalog from Addons

michaelKaefer committed:
Restructure and rename admin controller aread for modules catalog from Addons

michaelKaefer committed:
Bug fix

mickaelandrieu commented:
You're right... but as we have already published this page in 1.7.3 (I guess?) I don't know if we have the right to move the classes this way... ping @eternoendless.

Edit: to synchronize your repository with PrestaShop one, assuming the remote name of PrestaShop repository is "ps" and your own is "origin":

```
git checkout develop
git fetch ps
git rebase -i ps/develop
git push origin develop # now your repository is synced with our own
```

If you need the name of the remotes, try `git remote -v` :)

Edit2: don't be surprised I've just cleaned up your branch.

PierreRambaud committed:
Fix datepicker under VueJs

PierreRambaud committed:
assets

PierreRambaud committed:
Wrong selector for datepicker in combinaison page

PierreRambaud committed:
Missing dot for navigation menu

PierreRambaud committed:
Change wording

eternoendless committed:
Merge pull request #9133 from PierreRambaud/fix/datepicker

Regression with DatePicker and checkboxes on stock page (catalogue > stock)

eternoendless committed:
Build assets

PierreRambaud committed:
Fix 500 while getting order total

michaelKaefer commented:
@mickaelandrieu thank you!

lfluvisotto committed:
CO: The left and the right parts of assignment are equal

lfluvisotto committed:
CO: Case mismatch in class/function/method call

mickaelandrieu committed:
Merge pull request #9163 from lfluvisotto/case-mismatch-class-function-method-call

CO: Case mismatch in class/function/method call

mickaelandrieu committed:
Merge pull request #9139 from PierreRambaud/improvement/delivery-slip

Migrate Sell > Orders > Delivery Slips

mickaelandrieu committed:
Merge pull request #9105 from andypieters/develop

Check if key exists before checking the value

mickaelandrieu committed:
Merge pull request #9147 from fatmaBouchekoua/navigate-between-page

Navigate between pages in catalog page

mickaelandrieu committed:
Merge pull request #9127 from fouratachour/Product_page_button

Buttons in product page

mickaelandrieu committed:
Merge pull request #8956 from unlocomqx/customization

Add customization ID to cart ajax response and updateCart event data

eternoendless committed:
Merge pull request #8914 from sLorenzini/FF-169

Clean Header BO design for merchant

eternoendless committed:
Merge pull request #9160 from PierreRambaud/fix/boom-5659

Fix 500 while getting order total

Quetzacoalt91 committed:
Merge pull request #9162 from lfluvisotto/left-right-parts-assignment-equal

CO: The left and the right parts of assignment are equal

Commit:
TE: Update scripts on 1.7.4.x

eternoendless committed:
Update native modules

eternoendless committed:
Merge pull request #9167 from eternoendless/update-dependencies

Update native module versions

eternoendless committed:
Fix wrong translation domains

eternoendless committed:
Update translation catalog

PierreRambaud reviewed (changes requested):
The new name is redundant, `Modules\ModuleCatalog`, can be a simple `Modules\Catalog`. It's like having `Grid\GridRow`, `Grid\GridColumn` which can be renamed in `Grid\Row` and `Grid\Column`.
But I approved the restructuring! :+1:

Seleda committed:
index.php file for controllers/front/listing/ (#9128)

* FO: To prevent spam from contact form

* FO: index.php file for controllers/front/listing/

* FO: index.php file for controllers/front/listing/

tomlev committed:
fix product attachment list

tomlev committed:
fix smarty condition count

mickaelandrieu commented:
@PierreRambaud make sense! One more update @michaelKaefer :) 2 things are really hard in programming: managing the cache, and naming things 🤣

Commit:
use security and demo mode annotations in controller

tomlev committed:
Merge pull request #9170 from tomlev/fixConditions

Fix smarty condition count

mickaelandrieu committed:
fixed-namespace

Commit:
localization page improvements

Commit:
implement localization pack choice provider

mickaelandrieu committed:
Merge branch 'develop' into migrate-themecontroller

michaelKaefer committed:
Alphabetical order of routes list

michaelKaefer committed:
Merge remote-tracking branch 'upstream/develop' into develop

mickaelandrieu committed:
Merge pull request #9154 from michaelKaefer/migrate-themecontroller

Migrate Theme Catalog Page of Design Section

mickaelandrieu committed:
Merge pull request #9169 from tomlev/BOOM-5675

fix product attachment list

mickaelandrieu commented:
Edit: thinking about it, should we update [ThemeCatalogController](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php) to CatalogController into "Design" namespace @PierreRambaud?

PierreRambaud commented:
Yep good idea :)

mickaelandrieu commented:
One more "First contribution" issue then, don't do it!

mickaelandrieu committed:
Merge pull request #9171 from mickaelandrieu/change-namespace

Fixed namespace of Delivery Slips controller

michaelKaefer commented:
@mickaelandrieu What is a "First contribution" issue?

michaelKaefer committed:
Merge remote-tracking branch 'upstream/develop' into develop

michaelKaefer committed:
Add src/Core/Foundation/PrestaShopVersion

michaelKaefer committed:
Remove unused private method

Quetzacoalt91 committed:
Missing requirement message

Commit:
improve geolocation options

Commit:
update geolocation form error handling

Commit:
update geolocation routes & Link class

Commit:
remove legacy geolocation files

Commit:
remove unused namespaces

Commit:
move choice table styling to theme

michaelKaefer committed:
Fix

michaelKaefer committed:
Inject the version to constructor

michaelKaefer committed:
Renamed PrestaShopVersion to Version

michaelKaefer committed:
Check for valid versions with a regex

michaelKaefer committed:
Fix

michaelKaefer committed:
Put Version into service container (and remove dependency from Version)

mickaelandrieu committed:
Use the new service everywhere we can instead of the constant

eternoendless committed:
Merge pull request #9177 from hadrich-hatem/update_tests

Update scripts on 1.7.4.x

eternoendless committed:
Update translation catalogue

mickaelandrieu committed:
Merge pull request #9175 from Quetzacoalt91/requirement-message

Missing requirement message

PierreRambaud committed:
Merge branch '1.7.4.x' of github.com:PrestaShop/PrestaShop into merge-2-develop

PierreRambaud committed:
Missing change

Commit:
implement new validator

eternoendless committed:
Merge pull request #9182 from PierreRambaud/merge-2-develop

Merge 1.7.4.x in develop

Commit:
add reusable choice providers

Commit:
fixes

Commit:
update choice providers namepsace

Commit:
start workin on payment methods migration

Commit:
add payment module rendering

Commit:
show warning when theres no payment methods

Commit:
make page responsive

Commit:
handle multishop context

mickaelandrieu committed:
Fixed DataConfigurationInterface PHPDoc

Commit:
responsive fix

Quetzacoalt91 committed:
Merge pull request #9190 from mickaelandrieu/fix-interface-phpdocs

Fixed DataConfigurationInterface PHPDoc

michaelKaefer committed:
Merge remote-tracking branch 'upstream/develop' into develop

michaelKaefer committed:
Remove redundant name

michaelKaefer committed:
Remove redundant name

michaelKaefer committed:
Remove redundant name

michaelKaefer committed:
Remove redundant name

michaelKaefer commented:
@PierreRambaud @mickaelandrieu I removed the redundant name, thank you

michaelKaefer committed:
Rename method

michaelKaefer committed:
Rename method

michaelKaefer committed:
Rebase

michaelKaefer committed:
Rebase

michaelKaefer committed:
Method must be public

PierreRambaud commented on src/PrestaShopBundle/Controller/Admin/Improve/Modules/ModuleCatalogController.php:
```diff
@@ -24,7 +24,7 @@
  */
 
-namespace PrestaShopBundle\Controller\Admin\Improve;
+namespace PrestaShopBundle\Controller\Admin\Improve\Modules;
```
You forgot to rename file name :) still ModuleCatalogController.php :)

mickaelandrieu committed:
Delivery slips routing

paeddl committed:
BO: Fixed mixed up display in some cases at partial refunds

Quetzacoalt91 committed:
Typo

sarjon committed:
inject language id into choice provider instead of context

sarjon committed:
removed AdminGeolocationController from legacy controller tests

sarjon committed:
added test for GeolocationController index action

sarjon committed:
small fixes

michaelKaefer committed:
Move default value to top

mickaelandrieu committed:
Merge pull request #9150 from Quetzacoalt91/docker-test-e2e

Add dockerfile for single container run of E2E

michaelKaefer committed:
Rename file

michaelKaefer commented:
@PierreRambaud Thank you! I fixed this stupid mistake of mine!

mickaelandrieu committed:
Merge pull request #9066 from michaelKaefer/patch-5

Refactor Dispatcher::useDefaultController()

mickaelandrieu committed:
Make the service public

mickaelandrieu committed:
Merge pull request #9174 from michaelKaefer/core-foundation-version

Added a service to manage PrestaShop versions (may deprecate _PS_VERSION_)

mickaelandrieu committed:
Unused use statement for AppKernel

Commit:
update translation domains

Commit:
move payment module presenting to adapter service

Commit:
update docblocks

Commit:
update Link class & remove legacy AdminPaymentController

Commit:
use Version service in pack loader

PierreRambaud committed:
Merge pull request #9198 from mickaelandrieu/fixed-appkernel-call

Add use statement for AppKernel

PierreRambaud committed:
Merge branch 'improve/routing' of https://github.com/mickaelandrieu/PrestaShop into mickaelandrieu-improve/routing

PierreRambaud committed:
Update assets

Quetzacoalt91 committed:
Merge pull request #9137 from mickaelandrieu/improve/routing

Improve routing of PrestaShop application

sarjon committed:
move localization routes to appropriate location

sarjon committed:
rebuild assets

sarjon committed:
move geolocation routes to appropriate location

mickaelandrieu committed:
Merge pull request #9116 from sarjon/migrate-localization-page

Migrate localization page

mickaelandrieu committed:
Merge branch 'develop' into migrate/payment_method

Commit:
fix service naming

Commit:
address @PierreRambaud comments

Commit:
build assets

Commit:
fix form_rest

Commit:
WS: preserve position in category

concerning bug
http://forge.prestashop.com/browse/PSCSX-4218

keeps the previous position of a product in a category or adds it to the end of category if a new category was assigned

PierreRambaud committed:
Merge pull request #9097 from rdy4ever/fix_phone_filed_type

Fix phone fieldtype

mickaelandrieu committed:
Fixed coding styles

mickaelandrieu committed:
Merge pull request #9192 from sarjon/migrate/payment_method

Migrate Payment methods page

mickaelandrieu committed:
Merge pull request #9176 from sarjon/migrate/geolocation-page

Migrate "Improve > International > Localization > Geolocation" page

mbadrani committed:
Merge pull request #9131 from fatmaBouchekoua/export_shopping_carts

Add export shopping carts scenario

Commit:
remove old routing files

mickaelandrieu committed:
Merge pull request #9195 from paeddl/develop

Fixed mixed up display in some cases at partial refunds

mickaelandrieu committed:
Merge pull request #9204 from sarjon/remove-routing-files

Remove leftover routing files

mickaelandrieu committed:
Merge pull request #9202 from Shopmonauten/patch-11

Preserve position in category when adding a new Product using Web services

michaelKaefer committed:
Merge branch 'restructure-and-rename-admin-controller' of https://github.com/michaelKaefer/PrestaShop into restructure-and-rename-admin-controller

PierreRambaud commented:
Hum, look like there is something weird with commits list. Did you target the good branch?

michaelKaefer commented:
@PierreRambaud That was me. Hope I can fix it next monday!

PierreRambaud commented:
I'm closing this Pull request since the source is an unknow repository and this is in waiting for rebase since one year.
